apps/metric_vnode/src/metric_vnode.erl

```diff
@@ -474,27 +467,40 @@ do_put(Bucket, Metric, Time, Value,
 reply(Reply, {_, ReqID, _} = Sender, #state{node=N, partition=P}) ->
     riak_core_vnode:reply(Sender, {ok, ReqID, {P, N}, Reply}).
 
+%% The timestamp is primarily used by the vacuum to remove data in accordance
+%% with the bucket lifetime. The timestamp is only updated once per 
+%% vacuum cycle, and may not accurately reflect the current system time.
+%% Although more performant than the monotonic `now()', there are nevertheless
+%% performance penalties for calling this too frequently. Also, updating the
+%% time on a self tick message may prevent the VNode being marked as idle by
+%% riak core.
+timestamp() ->
+    erlang:system_time(milli_seconds).
+
+
 valid_ts(TS, Bucket, State) ->
-    case expiery(Bucket, State) of
+    case expiry(Bucket, State) of
         %% TODO:
         %% We ignore every data where the last point is older then the
         %% lifetime this means we could still potentially write in the
-        %% past if writing baches but this is a problem for another day!
+        %% past if writing batches but this is a problem for another day!
         {Exp, State1} when is_integer(Exp),
                            TS < Exp ->
             {false, State1};
         {_, State1} ->
             {true, State1}
     end.
+
 %% Return the latest point we'd ever want to save. This is more strict
 %% then the expiration we do on the data but it is strong enough for
 %% our guarantee.
-expiery(Bucket, State = #state{now = Now}) ->
+expiry(Bucket, State) ->
```

Comment:
this function should remain unchanged since `now` is still (or agian) in the `state` record